packages/apps/human-app/frontend/src/smart-contracts/chains.ts

```diff
@@ -38,13 +65,57 @@ export const MainnetChains: ChainWithAddresses[] = [
   {
     chainId: 137,
     name: 'Polygon',
-    rpcUrl: 'https://polygon-rpc.com/',
-    currency: 'MATIC',
     explorerUrl: 'https://polygonscan.com/',
+    rpcUrl: 'https://polygon-rpc.com/',
     addresses: MainnetContracts.Polygon,
   },
 ];
 
+export const AllTestnetsChains: ChainWithAddresses[] =
+  getChainsCfgByType<TestnetNarrow>([...testnetChains]);
+
+export const AllMainnetChains: ChainWithAddresses[] =
+  getChainsCfgByType<MainnetNarrow>([...mainnetChains]);
+
+function getChainsCfgByType<T extends TestnetNarrow | MainnetNarrow>(
+  chainsArr: T[]
+): ChainWithAddresses[] {
+  const initVal: ChainWithAddresses[] = [];
+
+  return chainsArr.reduce<ChainWithAddresses[]>((acc, currentValue) => {
```

Comment:
Why `reduce` and not just
```
for (const chain of chainsArr) {
 // logic
}
```
?

Reply:
@adrian-oleskiewicz simple `for` loop (or `.map`, or other primitive method) is always simpler to read/digest when you read the code, so less mind-fuel spent, so please - yes :)

Reply:
Fair enough :)